resource/graph_node.go

```diff
@@ -321,6 +326,20 @@ func (w *GraphNode) setNeedsReconfigure(newConfig Config, mustReconfigure bool,
 	w.unresolvedDependencies = dependencies
 }
 
+// UpdateRevision updates the node config revision if the node is in a [NodeStateReady]
+// state.
+func (w *GraphNode) UpdateRevision(revision string) {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+	if w.state == NodeStateReady {
+		w.config.Revision = revision
+		w.revision = revision
+		if w.logger != nil {
+			w.logger.Infow("updated revision", "new", revision)
```

Comment:
debug log instead? might be noisy otherwise

Reply:
good catch, I indeed just needed this debugging. I will probably just remove altogether.